tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/compose/HelmLegacyStatefulSetMigrationIT.java

```diff
@@ -0,0 +1,171 @@
+
+        // assert Control Center login
+        ControlCenterUtil.assertLogin(client,
+                operatorNamespace,
+                WEB_DRIVER_CONTAINER,
+                CC_SERVICE_NAME,
+                CC_SERVICE_PORT);
+        // publish retained messages
+        final var topicList =
+                MqttUtil.publishRetainedMessages(client, operatorNamespace, MQTT_SERVICE_NAME, MQTT_SERVICE_PORT);
+        // assert RestAPI service and authorization with ESE extension
+        RestAPIUtil.assertAuth(client, operatorNamespace, REST_API_SERVICE_NAME, REST_API_SERVICE_PORT);
+        // assert publishes metrics
+        MonitoringUtil.assertPublishesMetrics(client, operatorNamespace, METRICS_SERVICE_NAME, METRICS_SERVICE_PORT);
+
+        // scale down the legacy operator
+        // kubectl scale deployment <release-name>-hivemq-operator-operator --replicas=0 -n <namespace>
+        final var legacyOperatorDeploymentName = String.format("%s-hivemq-operator-operator", LEGACY_RELEASE_NAME);
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 0);
+        var legacyOperatorLabels = K8sUtil.getHiveMQLegacyOperatorLabels(LEGACY_RELEASE_NAME);
+        client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .waitUntilCondition(Objects::isNull, 1, TimeUnit.MINUTES);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()).isEmpty();
+
+        // delete custom resource with orphan
+        // kubectl delete hivemq-clusters.hivemq.com <release-name> --cascade=orphan -n <namespace>
+        final var legacyPlatform = K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        legacyPlatform.withPropagationPolicy(DeletionPropagation.ORPHAN).withTimeoutInMillis(5000).delete();
+        assertThat(K8sUtil.getLegacyHiveMQPlatform(client, operatorNamespace, LEGACY_RELEASE_NAME).get()).isNull();
+
+        // scale up the legacy operator
+        // kubectl scale deployment <release-name>-hivemq-operator-operator --replicas=1 -n <namespace>
+        K8sUtil.scaleDeployment(client, operatorNamespace, legacyOperatorDeploymentName, 1);
+        K8sUtil.waitForLegacyOperatorPodStateRunning(client, operatorNamespace, LEGACY_RELEASE_NAME);
+        assertThat(client.pods()
+                .inNamespace(operatorNamespace)
+                .withLabels(legacyOperatorLabels)
+                .list()
+                .getItems()
+                .size()).isEqualTo(1);
+
+        //TODO: We cannot execute `helm uninstall my-release --cascade orphan` - See: https://github.com/helm/helm/issues/13279
+        // we need to uninstall the legacy release with orphan option so the new platform Helm install takes over existing resources
```

Comment:
Also a tiny bit of context would be great (without the need to read the linked issue). Maybe something like:
```suggestion
        // we need to uninstall the legacy release with orphan option so the new platform Helm install takes over existing resources
        //TODO: We cannot execute `helm uninstall my-release --cascade orphan`, because this will delete resources
        // without an owner reference, that are still needed - See: https://github.com/helm/helm/issues/13279
```